Add unit test for secure_make_dirs() to handle nested dirs

avilcheslopez opened the pull request:
Ensure secure_make_dirs() can create a requested path when parent directories don't exist.

- Fixes #2906

avilcheslopez committed:
Add unit test for secure_make_dirs() to handle nested dirs

Ensure secure_make_dirs() can create a requested path when parent
directories don't exist.

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

bgeltz commented on service/geopmdpy_test/TestSecureFiles.py:
```diff
@@ -246,6 +249,26 @@ def test_creation_bad_group_owner(self):
+        os.umask(0o000)
+        perm_mode = 0o711
+        secure_make_dirs(sess_path, perm_mode)
+        self.check_dir_perms(sess_path, perm_mode)
```
```suggestion
        self.check_dir_perms(sess_path, perm_mode)
        self.check_dir_perms(base_path, perm_mode)
```
Maybe this is redundant with what you have below?

avilcheslopez replied:
I don't believe so. The check below ensures the base directory doesn't get any of the denied permissions set by the umask `secure_make_dirs()` is supposed to set.

Perhaps I should just mock out `os.umask()` and verify `secure_make_dirs()` calls it with the right parameter.

avilcheslopez committed:
secure_make_dirs() unit test improvements

- Ensure umask is restored
- Check for correct permissions for nested directory creation

Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

cmcantalupo reviewed (approved):
LGTM
